clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -311,7 +312,7 @@ public void handshake() throws IOException {
 
         } catch (SSLException e) {
             handshakeFailure();
-            throw e;
+            throw new AuthenticationException("SSL handshake failed", e);
```

Comment:
Is it intentional that we're using `javax.security.sasl.AuthenticationException` here?

Reply:
No, I have removed that. Will be done properly by the SSL authentication PR.